core/src/main/java/org/apache/calcite/sql/SqlBasicTypeNameSpec.java

```diff
@@ -245,16 +249,31 @@ private static boolean isWithLocalTimeZoneDef(SqlTypeName typeName) {
     }
   }
 
+  private static boolean isWithTimeZoneDef(SqlTypeName typeName) {
```

Comment:
Alternatively you could add a set/list to `SqlTypeName` (see `DATETIME_TYPES` for example) and then just use `.contains()` or something similar.